packages/core/src/routes/interaction/verifications/mandatory-user-profile-validation.ts

```diff
@@ -75,28 +84,107 @@ const getMissingProfileBySignUpIdentifiers = ({
   return missingProfile;
 };
 
-// This is a fallback logic make sure the user has a valid identifier for register should be guarded by the SIE already
+// This is a fallback logic make sure the user has a valid identifier for sign-up. Should be guarded by the SIE already
 const validateRegisterMandatoryUserProfile = (profile?: Profile) => {
   assertThat(
     profile && (profile.username ?? profile.email ?? profile.phone ?? profile.connectorId),
     new RequestError({ code: 'user.missing_profile', status: 422 })
   );
 };
 
+// Fill the missing email or phone from the social identity if any
+const fillMissingProfileWithSocialIdentity = async (
+  missingProfile: Set<MissingProfile>,
+  interaction: MandatoryProfileValidationInteraction,
+  userQueries: Queries['users']
+): Promise<[Set<MissingProfile>, MandatoryProfileValidationInteraction]> => {
+  const { identifiers = [], profile } = interaction;
+
+  const socialIdentifier = identifiers.find(
+    (identifier): identifier is SocialIdentifier => identifier.key === 'social'
+  );
+
+  if (!socialIdentifier) {
+    return [missingProfile, interaction];
+  }
+
+  const {
+    userInfo: { email, phone },
+  } = socialIdentifier;
+
+  if (
+    (missingProfile.has(MissingProfile.email) || missingProfile.has(MissingProfile.emailOrPhone)) &&
+    email &&
+    // Email taken
+    !(await userQueries.hasUserWithEmail(email))
+  ) {
+    missingProfile.delete(MissingProfile.email);
+    missingProfile.delete(MissingProfile.emailOrPhone);
+
+    // Assign social verified email to the interaction
+    return [
+      missingProfile,
+      {
+        ...interaction,
+        identifiers: mergeIdentifiers({ key: 'emailVerified', value: email }, identifiers),
+        profile: {
+          ...profile,
+          email,
+        },
+      },
+    ];
+  }
```

Comment:
What if the phone is also available in social identify? Should update both before returning?

Reply:
checked, currently, we do not have both existing cases. If so, we might have to update the flow as well.  Strictly saying if only email is required by our SignUp settings, we should only fill in the email as well.